module/signature/aggregation.go

```diff
@@ -158,22 +158,19 @@ func (s *SignatureAggregatorSameMessage) HasSignature(signer int) (bool, error)
 	return ok, nil
 }
 
-// Aggregate aggregates the stored BLS signatures and returns the aggregated signature.
-//
-// Aggregate attempts to aggregate the internal signatures and returns the resulting signature.
-// The function errors if any signature fails the deserialization. It also performs a final
-// verification and errors if the aggregated signature is not valid.
-// It also errors if no signatures were added.
-// Post-check of aggregated signature is required for function safety, as `TrustedAdd` allows
-// adding invalid signatures. Aggregation may also output an invalid signature (identity)
-// even though all included signatures are valid (extremely unlikely case when all keys are sampled
-// uniformly)
+// Aggregate aggregates the stored BLS signatures and returns the resulting aggregated signature.
+// The method errors if no signatures were added or any of the added signature fails deserialization.
+// It also performs a final verification and errors if the aggregated signature is invalid. This
+// "post-check" of the aggregated signature is required for safety, as `TrustedAdd` allows
+// adding invalid signatures.
 // The function is not thread-safe.
 // Returns:
 //   - InsufficientSignaturesError if no signatures have been added yet
-//   - InvalidAggregatedSignatureError if the aggregated signature is invalid. It's not clear whether included
-//     signatures via TrustedAdd are invalid. This case can happen even when all added signatures
-//     are individually valid.
+//   - ErrIdentitySignature if aggregation produced the identity signature, which is invalid
+//     by convention. This error can arise in two scenarios:
```

Comment:
I would remove "by convention" so that we don't imply that only this module made the convention, while the invalidity of the identity signature comes from the low level crypto layer. 

```suggestion
//     This error can arise in two scenarios:
```

Reply:
done